### ISSUE TITLE
DBeaver holding transaction to Postgres Database in AutoCommit even after closing Query window

### ISSUE THREAD

**Issue by barrelmonkey** [dbeaver/dbeaver#7402]
#### System information: 
- Operating system (distribution) and version: Mac OSX Catalina
- DBeaver version: 6.2.5.201911171645
- Java version: 1.8.0_91 
- Additional extensions

#### Connection specification:
- Database name and version: Postgres 10.9
- Driver name: PostgreSQL Native Driver PostgreSQL 9.4.1212.jre7
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
I have my window configured for Auto Commit. It shows Auto in the box at the top. The pending transactions list is empty, but the database pg_stat_activity table shows an 'idle in transaction' process. Even after closing the query window I can still see the process as 'idle in transaction'. The application name even has the tab name in there, but that tab is no longer open. The only way to get rid of the connection is by completely disconnecting from the database. We haven't noticed this until recently, and it doesn't appear to be every single time.

#### Steps to reproduce, if exist:
Connect to Database
Open Query Window
Ensure Commit Mode is 'Auto'
Run a select query
In a separate window or ideally separate connection client check the pg_stat_activity records. It should show the original window as 'idle in transaction'.
Close the original window. The record should continue to show.
Drop the database connection. The record should disappear.

**Comment by serge-rider** [comment]
What is the problem with this pending transactions? It doesn't lock any resources as far as I understood.
Any open connection which had some activity looks like this, at least if it was initiated thru jdbc.
https://stackoverflow.com/questions/51019/what-does-it-mean-when-a-postgresql-process-is-idle-in-transaction

DBeaver opens connection once you initiated it and closes one you explicitly close it in DBeaver UI. 
We don't close connections implicitly.
So this doesn't looks like a bug or a problem for me. But maybe I'm, wrong. Anyway I don't see what we can fix here..

**Comment by barrelmonkey** [comment]
If it is normal behaviour then it would be seen more consistently, but we don't. The things that lead me to believe this is a bug are as follows.

1. We have never seen this until recently, suggesting it is something recently changed
2. Even when closing my query window the transaction remains open, which surely shouldn't be the case?

There is a difference between having an open connection which may have a state of 'idle' whilst it isn't doing anything, and having an open connection that is 'idle in transaction' which means it is in the middle of doing something within a transaction and waiting for further processing, either for user interaction in a transaction, or possible because it is waiting on a lock being released.

Having these 'idle in transaction' connections open can have adverse affects, for example it can prevent RDS replication if you are connected to a replication database (I don't know how, I just know that it does).
From a production database viewpoint it also blocks anything that requires an ACCESS EXCLUSIVE lock, for example any REINDEX, VACUUM, REFRESH MATERIALIZED VIEW, along with any DROP or ALTER queries on the tables.

**Comment by serge-rider** [comment]
@barrelmonkey please test it in 6.3.2 version. 
I can't reproduce this issue anymore.

**Comment by tony-sappe** [comment]
Either this issue resurfaced or it wasn't fully resolved. The reported issue has caused problems as described in previous comments. The best way we've found to prevent this behavior is by disabling auto-commit mode in DBeaver.

See example db statement:

![image](https://user-images.githubusercontent.com/22781949/93113077-f88fc680-f675-11ea-9c3a-97b941b40e9b.png)

Env similar to original issue post:
MacOS Catalina
Postgres 10.12
DBeaver 7.1.0

**Comment by TM818** [comment]
I'm having an issue with DBeaver leaving an open transaction, causing a lock, that's preventing me from running DB migrations (creating a new table, changing a column name, etc). The DB migrations just sit there frozen until I cycle postgres (systemctl restart postgresql).

Like the other users here, I'm running postgres:

Amazon Linux 2 
Postgres 10.4
DBeaver 7.1.5

I am running DBeaver 7.1.5 on a Mac 10.15.6 (the database in question is on a remote EC2 host). 

I have confirmed DBeaver is the issue by querying the pg_stat_activity table (as pg root) and seeing that the transaction causing the lock is DBeaver, with the state left at "idle in transaction" indefinitely (in this example, it was 5 minutes since I touched the DBeaver GUI - it was freshly opened, and I did not do anything with DBeaver but view a database table in the GUI - I did not request any changes / transactions to the database). 

Here is example output from pg_stat_activity of the DBeaver process causing the lock simply by the client accessing the DB: 

-[ RECORD 3 ]----+-----------------------------------------------------------------------------------------------------------------------
datid            | 16384
datname          | ******
pid              | 14540
usesysid         | 16385
usename          | ******
application_name | DBeaver 7.1.5 - Main <*****>
client_addr      | 185.92.26.210
client_hostname  | 
client_port      | 49586
backend_start    | 2020-09-24 19:56:39.773692+00
xact_start       | 2020-09-24 19:56:41.09699+00
query_start      | 2020-09-24 19:56:57.375199+00
state_change     | 2020-09-24 19:56:57.37917+00
wait_event_type  | Client
wait_event       | ClientRead
state            | idle in transaction
backend_xid      | 
backend_xmin     | 2744
query            | select c.oid,pg_catalog.pg_total_relation_size(c.oid) as total_rel_size,pg_catalog.pg_relation_size(c.oid) as rel_size+
                 | FROM pg_class c                                                                                                       +
                 | WHERE c.relnamespace=$1
backend_type     | client backend

**Comment by uslss** [comment]
seems issue is fixed in 7.3.0.
If it is still actual - please create new issue.

**Comment by zachsa** [comment]
I am using 7.3.2

In a SQL editor with a simple query:

```sql
select 1;
```

I get an error 

```
SQL Error [25P02]: ERROR: current transaction is aborted, commands ignored until end of transaction block
```

I have to create a new SQL editor otherwise I can't query the database at all without that error (this is after experimenting with a transaction in the same editor)
